Closes #1972 intial gsm.mapping spin out

zdz2101 opened the pull request:
## Overview
The following functions were deleted and migrated to [gsm.mapping](https://github.com/Gilead-BioStats/gsm.mapping)

- ApplySpec()
- CalculatePercentage()
- CombineSpecs()
- Ingest()
- MakeLongMeta()
- RunQuery()

zdz2101 committed:
see what breaks

zdz2101 committed:
move tests over

zdz2101 committed:
this branch should work now

zdz2101 committed:
export the stop_if

zdz2101 committed:
typo

zdz2101 committed:
praying

zdz2101 committed:
what does that do

zdz2101 committed:
undo that

zdz2101 committed:
does that help

zdz2101 committed:
Merge branch 'dev' into fix-1972v2

zdz2101 committed:
gsm.mapping is just a suggest

zdz2101 committed:
Merge branch 'fix-1972v2' of https://github.com/Gilead-BioStats/gsm into fix-1972v2

zdz2101 committed:
what does that do

zdz2101 committed:
does that do it?

zdz2101 committed:
is it an actions issue

zdz2101 committed:
seems this is necessary?

zdz2101 committed:
give it a ref

zdz2101 committed:
if we can get a PAT into repo secrets see if this cicd works

zdz2101 committed:
Merge branch 'dev' into fix-1972v2

zdz2101 committed:
typo

zdz2101 committed:
Merge branch 'fix-1972v2' of https://github.com/Gilead-BioStats/gsm into fix-1972v2

zdz2101 committed:
circular logic it seems?

zdz2101 committed:
hopefully that passes

zdz2101 committed:
fix warning

zdz2101 committed:
merge conflict

Merge branch 'dev' of https://github.com/Gilead-BioStats/gsm into fix-1972v2

# Conflicts:
#	R/util-CalculatePercentage.R
#	R/util-Ingest.R
#	man/CalculatePercentage.Rd
#	tests/testthat/test-RunWorkflow.R

zdz2101 committed:
last conflict

zdz2101 committed:
undo mistake

zdz2101 committed:
missing some deletions

zdz2101 committed:
use for dev PRs so far

zdz2101 committed:
fix last two ci/cd errors

zdz2101 committed:
forgot to add stop_if

lauramaxwell commented on man/glossary/dfGroups.yaml:
```diff
@@ -1,5 +1,5 @@
 definition: >
   Group-level metadata dictionary. Created by passing CTMS
-  site and study data to [MakeLongMeta()]. Expected columns:
+  site and study data to [gsm.mapping::MakeLongMeta()]. Expected columns:
```
nice catch!

lauramaxwell commented on .github/workflows/pkgdown-pr.yaml:
```diff
@@ -19,6 +19,7 @@ jobs:
 
     env:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      REPOPAT: ${{ secrets.GSMMAPPING }}
```
this is good for now, but i'm wondering if we can just replace the existing `GITHUB_TOKEN` with one that has the necessary permissions so we only need one PAT that can be used throughout, thereby removing the need for the "Install private repos with remotes" step below.

zdz2101 replied:
I tried swapping `secrets.GITHUB_TOKEN` with `secrets.GSMMAPPING` in a separate instances and the whole thing just kinda bricks and falls apart, maybe I was tinkering with more than just one individual thing at a time

lauramaxwell reviewed (commented):
This looks great! I think we should probably be removing the `inst/workflow/1_mappings` directory from this package and updating all the examples to refer to `gsm.mapping` in `MakeWorkflowList()` mapping calls. we could also do that in another PR, to keep this one a bit more manageable. Let me know what you think and I'll approve/request changes based on the discussion

zdz2101 commented:
> This looks great! I think we should probably be removing the `inst/workflow/1_mappings` directory from this package and updating all the examples to refer to `gsm.mapping` in `MakeWorkflowList()` mapping calls. we could also do that in another PR, to keep this one a bit more manageable. Let me know what you think and I'll approve/request changes based on the discussion

My only concern is there is sprinkles of needing the mapped data across testing, vignettes, and inst/examples, so I'm curious how you would "access" these if you remove them? a gsm.template style solution that copies it over?

jonthegeek commented:
> This looks great! I think we should probably be removing the `inst/workflow/1_mappings` directory from this package and updating all the examples to refer to `gsm.mapping` in `MakeWorkflowList()` mapping calls. we could also do that in another PR, to keep this one a bit more manageable. Let me know what you think and I'll approve/request changes based on the discussion

Be careful about circular references. {gsm.mapping} currently imports {gsm}, so using {gsm.mapping} inside {gsm} can lead to strangeness. I *think* we'll be fine for now at least with {gsm.mapping} in Suggests of {gsm}, but, if we're aiming for CRAN eventually, I don't think it'll fly.

zdz2101 commented:
> > This looks great! I think we should probably be removing the `inst/workflow/1_mappings` directory from this package and updating all the examples to refer to `gsm.mapping` in `MakeWorkflowList()` mapping calls. we could also do that in another PR, to keep this one a bit more manageable. Let me know what you think and I'll approve/request changes based on the discussion
> 
> Be careful about circular references. {gsm.mapping} currently imports {gsm}, so using {gsm.mapping} inside {gsm} can lead to strangeness. I _think_ we'll be fine for now at least with {gsm.mapping} in Suggests of {gsm}, but, if we're aiming for CRAN eventually, I don't think it'll fly.

Yes I encountered a bit of this already, the combination of the packages/use cases to work together needs to be the most downstream. I've worked with organizations that have one repo that spins out the testing/using of multiple packages that work together into a separate repo as well, something like `gsm.testing` that will depend on core, mapping, kri, etc.

Perhaps a meeting for design needs to be had?

lauramaxwell commented:
> > Be careful about circular references. {gsm.mapping} currently imports {gsm}, so using {gsm.mapping} inside {gsm} can lead to strangeness. I _think_ we'll be fine for now at least with {gsm.mapping} in Suggests of {gsm}, but, if we're aiming for CRAN eventually, I don't think it'll fly.
> 
> Yes I encountered a bit of this already, the combination of the packages/use cases to work together needs to be the most downstream. I've worked with organizations that have one repo that spins out the testing/using of multiple packages that work together into a separate repo as well, something like `gsm.testing` that will depend on core, mapping, kri, etc.
> 
> Perhaps a meeting for design needs to be had?

Ah good points, both. ok, i definitely think a design meeting is in order. now that we have this first step towards breaking these apart working, I think it brings a lot more issues to light, and will make this discussion easier and more useful. I'll schedule something for next week when Jeremy is back, since i'm sure he would like to be involved. Happy to meet this week to gather issues/problems to address in the meantime.

All that being said, this works well and I think accomplishes what we were hoping to accomplish with the usage of gsm.mapping 0.0.1, so unless @jonthegeek has additional comments/suggestions, I think i'm ok to merge at this time.

jonthegeek commented:
> All that being said, this works well and I think accomplishes what we were hoping to accomplish with the usage of gsm.mapping 0.0.1, so unless @jonthegeek has additional comments/suggestions, I think i'm ok to merge at this time.

Nope! I'm just on "avoid circular references!" patrol for this, no comments on the functionality here 😁

zdz2101 commented:
> > > This looks great! I think we should probably be removing the `inst/workflow/1_mappings` directory from this package and updating all the examples to refer to `gsm.mapping` in `MakeWorkflowList()` mapping calls. we could also do that in another PR, to keep this one a bit more manageable. Let me know what you think and I'll approve/request changes based on the discussion
> > 
> > 
> > Be careful about circular references. {gsm.mapping} currently imports {gsm}, so using {gsm.mapping} inside {gsm} can lead to strangeness. I _think_ we'll be fine for now at least with {gsm.mapping} in Suggests of {gsm}, but, if we're aiming for CRAN eventually, I don't think it'll fly.
> 
> What i'm suggesting would just be to change the calls to `MakeWorkflowList(strPath = "workflow/1_mapping", package = "gsm.mapping")`, in the examples, not to reference any gsm.mapping in the gsm codebase. Either way, I understand the point here, and we'll need to sort this out.

I'm okay with doing this removal and MakeWorkflowList as a separate PR too so this doesnt grow too unruly

lauramaxwell commented:
> > > > This looks great! I think we should probably be removing the `inst/workflow/1_mappings` directory from this package and updating all the examples to refer to `gsm.mapping` in `MakeWorkflowList()` mapping calls. we could also do that in another PR, to keep this one a bit more manageable. Let me know what you think and I'll approve/request changes based on the discussion
> > > 
> > > 
> > > Be careful about circular references. {gsm.mapping} currently imports {gsm}, so using {gsm.mapping} inside {gsm} can lead to strangeness. I _think_ we'll be fine for now at least with {gsm.mapping} in Suggests of {gsm}, but, if we're aiming for CRAN eventually, I don't think it'll fly.
> > 
> > 
> > What i'm suggesting would just be to change the calls to `MakeWorkflowList(strPath = "workflow/1_mapping", package = "gsm.mapping")`, in the examples, not to reference any gsm.mapping in the gsm codebase. Either way, I understand the point here, and we'll need to sort this out.
> 
> I'm okay with doing this removal and MakeWorkflowList as a separate PR too so this doesnt grow too unruly

You can make an issue for this, but i would hold off on it until after we meet with the whole team about the larger vision for this package restructure.

lauramaxwell reviewed (approved):
LGTM 👍

lauramaxwell committed:
add gsm.mapping requirement to `3_ReportingWorkflow` example

zdz2101 committed:
swap RunQuery back in

zdz2101 commented:
@lauramaxwell okay looks like the migration back succeeded

lauramaxwell commented:
@jwildfire

zdz2101 commented:
@jwildfire @lauramaxwell instead of merging to dev, are we still going to have a fork or separate branch that contains all this?

zdz2101 committed:
migrate the 3_reporting functions

zdz2101 committed:
see if the ci/cd allows this separation

zdz2101 committed:
roxygen stuff dependent on some of the new function descriptions

zdz2101 committed:
see if this works

zdz2101 committed:
pretty sure that still does nothing

zdz2101 committed:
why does this work?

zdz2101 committed:
omg im a moron

zdz2101 committed:
change it back to previous

zdz2101 committed:
catch up to dev

Merge branch 'dev' into fix-1972v2

# Conflicts:
#	man/MakeCharts.Rd
#	man/MakeMetricTable.Rd
#	man/MakeParamLabels.Rd
#	man/MakeStudyInfo.Rd
#	man/MakeWideGroups.Rd
#	man/Report_KRI.Rd
#	man/Report_MetricTable.Rd
#	man/Report_Setup.Rd
#	man/Report_StudyInfo.Rd
#	man/Visualize_Metric.Rd
#	man/Widget_BarChart.Rd
#	man/Widget_GroupOverview.Rd
#	man/Widget_ScatterPlot.Rd
#	man/Widget_TimeSeries.Rd
#	man/shared-params.Rd

zdz2101 committed:
fix local check warning, why isn't cicd playing nicely anymore

zdz2101 committed:
what happened

zdz2101 committed:
catch up to fix-1972v2

Merge branch 'fix-1972v2' into fix-2010

# Conflicts:
#	man/MakeBounds.Rd
#	man/MakeCharts.Rd
#	man/MakeMetricTable.Rd
#	man/Report_FlagOverTime.Rd
#	man/Report_KRI.Rd
#	man/Report_MetricTable.Rd
#	man/Report_OverviewText.Rd
#	man/Report_Setup.Rd
#	man/Visualize_Metric.Rd
#	man/Visualize_Scatter.Rd
#	man/Visualize_Score.Rd
#	man/Widget_BarChart.Rd
#	man/Widget_FlagOverTime-shiny.Rd
#	man/Widget_FlagOverTime.Rd
#	man/Widget_GroupOverview.Rd
#	man/Widget_ScatterPlot.Rd
#	man/Widget_TimeSeries.Rd
#	man/shared-params.Rd

zdz2101 committed:
fix merge conflict stuff

zdz2101 committed:
missed a few

zdz2101 committed:
why can't it install like it used to

zdz2101 committed:
PLEASE

zdz2101 committed:
okay please

zdz2101 committed:
okay please

zdz2101 committed:
fix roxygen stuff

zdz2101 committed:
cleanup vignettes a bit

zdz2101 committed:
fix more roxygen stuff

zdz2101 commented:
@lauramaxwell Once #2018 looks good, all we need to do here is modify the `.github` files to point at the correct branch of `gsm.mapping` (should be dev, currently points to `fix-5_6` which will contain all the new additional changes, once `fix-5_6` is completed and reviewed we can make the modification)

lauramaxwell committed:
change checks back to dev PRs

lauramaxwell committed:
Merge pull request #2018 from Gilead-BioStats/fix-2010

Closes #2010 migration of additional mapping aspects

lauramaxwell commented:
Closing this in favor of #2031

zdz2101 commented on .github/workflows/R-CMD-check-dev.yaml:
```diff
@@ -22,6 +22,7 @@ jobs:
 
     env:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      REPOPAT: ${{ secrets.GSMMAPPING }}
```
@samussiah starts around here, a repo pat needs to be added

zdz2101 commented on .github/workflows/R-CMD-check-dev.yaml:
```diff
@@ -37,9 +38,14 @@ jobs:
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          extra-packages: any::rcmdcheck
+          extra-packages: any::rcmdcheck, gsm.mapping=?ignore
```
you ignore the private package in question in the set r dependencies

zdz2101 commented on .github/workflows/R-CMD-check-dev.yaml:
```diff
@@ -37,9 +38,14 @@ jobs:
+      - name: Install private repos with remotes
+        run: |
+          Rscript -e 'install.packages("remotes")'
+          Rscript -e 'remotes::install_github("Gilead-BioStats/gsm.mapping", ref = "fix-5_6", auth_token = Sys.getenv("REPOPAT"))'
```
manually install the missing/private package in the dependencies as appropriate

zdz2101 commented on .github/workflows/R-CMD-check-dev.yaml:
```diff
@@ -37,9 +38,14 @@ jobs:
+          Rscript -e 'install.packages("remotes")'
+          Rscript -e 'remotes::install_github("Gilead-BioStats/gsm.mapping", ref = "fix-5_6", auth_token = Sys.getenv("REPOPAT"))'
+
       - uses: r-lib/actions/check-r-package@v2
```
then run the usual check

zdz2101 reviewed (commented):
@samussiah
